--- file-5.05/magic/Makefile.am~	2011-01-18 19:19:39.000000000 +0200
+++ file-5.05/magic/Makefile.am	2011-01-18 19:21:23.958686512 +0200
@@ -5,11 +5,14 @@
 MAGIC_DIR = $(top_srcdir)/magic
 MAGIC_FRAGMENT_DIR = $(MAGIC_DIR)/$(MAGIC_FRAGMENT_BASE)
 
-pkgdata_DATA = magic.mgc
+pkgdata_DATA = magic.mgc magic
 
 EXTRA_DIST = \
 $(MAGIC_DIR)/Header \
 $(MAGIC_DIR)/Localstuff \
+$(MAGIC_FILES)
+
+MAGIC_FILES = \
 $(MAGIC_FRAGMENT_DIR)/acorn \
 $(MAGIC_FRAGMENT_DIR)/adi \
 $(MAGIC_FRAGMENT_DIR)/adventure \
@@ -236,8 +239,9 @@
 $(MAGIC_FRAGMENT_DIR)/dump \
 $(MAGIC_FRAGMENT_DIR)/zyxel 
 
-MAGIC = magic.mgc
-CLEANFILES = ${MAGIC} $(MAGIC_FRAGMENT_DIR)/Localstuff
+MAGIC = magic
+MAGIC_MGC = magic.mgc
+CLEANFILES = $(MAGIC) $(MAGIC_MGC) $(MAGIC_FRAGMENT_DIR)/Localstuff
 
 # FIXME: Build file natively as well so that it can be used to compile
 # the target's magic file
@@ -253,8 +253,12 @@
 FILE_COMPILE_DEP = $(FILE_COMPILE)
 endif
 
-${MAGIC}: $(EXTRA_DIST) $(FILE_COMPILE_DEP)
-	@rm -fr magic
-	@mkdir magic && cp -p $(EXTRA_DIST) magic
-	$(FILE_COMPILE) -C -m magic
-	@rm -fr magic
+$(MAGIC): $(MAGIC_DIR)/Header $(MAGIC_FILES)
+	cat $(MAGIC_DIR)/Header $(MAGIC_FILES) > $@
+
+$(MAGIC_MGC): $(MAGIC) $(FILE_COMPILE_DEP)
+	$(FILE_COMPILE) -C -m $<
+
+install-data-local:
+	install -d $(DESTDIR)$(sysconfdir)
+	cp -p $(MAGIC_DIR)/Localstuff $(DESTDIR)$(sysconfdir)/magic
