--- file-3.37/magic.mime.orig	2002-04-24 13:05:20.000000000 -0300
+++ file-3.37/magic.mime	2002-04-24 13:06:18.000000000 -0300
@@ -517,13 +517,24 @@
 >4      byte            0
 >4      byte            1
 >4      byte            2
+# offset 5: data encoding:
+# invalid data encoding
 >5      byte            0
+# little endian
 >5      byte            1
 >>16    leshort         0
 >>16    leshort         1               application/x-object
 >>16    leshort         2               application/x-executable
 >>16    leshort         3               application/x-sharedlib
 >>16    leshort         4               application/x-coredump
+# big endian:
+>5      byte            2
+>>16    beshort         0
+>>16    beshort         1               application/x-object
+>>16    beshort         2               application/x-executable
+>>16    beshort         3               application/x-sharedlib
+>>16    beshort         4               application/x-coredump
+
 #
 # DOS
 0		string			MZ				application/x-dosexec
