--- file-4.02/magic/Magdir/policy.selinux	2003-08-28 14:36:54.000000000 -0400
+++ file-4.02/magic/Magdir/policy	2003-08-28 14:39:35.000000000 -0400
@@ -0,0 +1,6 @@
+# SE Linux policy database
+0       lelong  0xf97cff8c      SE Linux policy
+>16     lelong  x               v%d
+>20     lelong  1 		MLS
+>24     lelong  x       	%d symbols
+>28     lelong  x       	%d ocons
--- file-4.02/magic/Makefile.am.selinux	2003-08-28 14:41:58.000000000 -0400
+++ file-4.02/magic/Makefile.am	2003-08-28 14:45:15.000000000 -0400
@@ -135,6 +135,7 @@
 Magdir/pgp \
 Magdir/pkgadd \
 Magdir/plus5 \
+Magdir/policy \
 Magdir/printer \
 Magdir/project \
 Magdir/psdbms \
