--- file-4.25/magic/Magdir/msdos.orig	2008-07-19 21:53:45.598043088 +0200
+++ file-4.25/magic/Magdir/msdos	2008-07-19 22:14:11.858041412 +0200
@@ -5,14 +5,13 @@
 
 # .BAT files (Daniel Quinlan, quinlan@yggdrasil.com)
 # updated by Joerg Jenderek
-0	string	@			
->1	string/cB	\ echo\ off	MS-DOS batch file text
+0	string/cB	@\ echo\ off	MS-DOS batch file text
 !:mime	text/x-msdos-batch
->1	string/cB	echo\ off	MS-DOS batch file text
+0	string/cB	@echo\ off	MS-DOS batch file text
 !:mime	text/x-msdos-batch
->1	string/cB	rem\ 		MS-DOS batch file text
+0	string/cB	@rem\ 		MS-DOS batch file text
 !:mime	text/x-msdos-batch
->1	string/cB	set\ 		MS-DOS batch file text
+0	string/cB	@set\ 		MS-DOS batch file text
 !:mime	text/x-msdos-batch
 
 
@@ -322,11 +321,12 @@
 #>0x20e string		SFX\ by\ LARC	\b, LARC self-extracting archive
 #0	byte		0xb8		COM executable
 # modified by Joerg Jenderek
->1	lelong		!0x21cd4cff	for DOS
+#>1	lelong		!0x21cd4cff	for DOS
 # http://syslinux.zytor.com/comboot.php
 # (32-bit COMBOOT) programs *.C32 contain 32-bit code and run in flat-memory 32-bit protected mode
 # start with assembler instructions mov eax,21cd4cffh
->1	lelong		0x21cd4cff	(32-bit COMBOOT)
+#>1	lelong		0x21cd4cff	(32-bit COMBOOT)
+0	string	\xb8\xff\x4c\xcd\x21	COM executable (32-bit COMBOOT)
 0	string	\x81\xfc		
 >4	string	\x77\x02\xcd\x20\xb9	
 >>36	string	UPX!			FREE-DOS executable (COM), UPX compressed
@@ -334,27 +334,27 @@
 # GRR search is not working
 #2	search/28	\xcd\x21	COM executable for MS-DOS
 #WHICHFAT.cOM
-2	string	\xcd\x21		COM executable for DOS
+#2	string	\xcd\x21		COM executable for DOS
 #DELTREE.cOM DELTREE2.cOM
-4	string	\xcd\x21		COM executable for DOS
+#4	string	\xcd\x21		COM executable for DOS
 #IFMEMDSK.cOM ASSIGN.cOM COMP.cOM
-5	string	\xcd\x21		COM executable for DOS
+#5	string	\xcd\x21		COM executable for DOS
 #DELTMP.COm HASFAT32.cOM
-7	string	\xcd\x21		
->0	byte	!0xb8			COM executable for DOS
+#7	string	\xcd\x21		
+#>0	byte	!0xb8			COM executable for DOS
 #COMP.cOM MORE.COm
-10	string	\xcd\x21		
->5	string	!\xcd\x21		COM executable for DOS
+#10	string	\xcd\x21		
+#>5	string	!\xcd\x21		COM executable for DOS
 #comecho.com
-13	string	\xcd\x21		COM executable for DOS
+#13	string	\xcd\x21		COM executable for DOS
 #HELP.COm EDIT.coM
-18	string	\xcd\x21		COM executable for MS-DOS
+#18	string	\xcd\x21		COM executable for MS-DOS
 #NWRPLTRM.COm
-23	string	\xcd\x21		COM executable for MS-DOS
+#23	string	\xcd\x21		COM executable for MS-DOS
 #LOADFIX.cOm LOADFIX.cOm
-30	string	\xcd\x21		COM executable for MS-DOS
+#30	string	\xcd\x21		COM executable for MS-DOS
 #syslinux.com 3.11
-70	string	\xcd\x21		COM executable for DOS
+#70	string	\xcd\x21		COM executable for DOS
 # many compressed/converted COMs start with a copy loop instead of a jump
 0x6	search/0xa	\xfc\x57\xf3\xa5\xc3	COM executable for MS-DOS
 0x6	search/0xa	\xfc\x57\xf3\xa4\xc3	COM executable for DOS
