--- file-4.02/magic/Magdir/policy.selinux	2003-08-28 14:36:54.000000000 -0400
+++ file-4.02/magic/Magdir/policy	2003-08-28 14:39:35.000000000 -0400
@@ -0,0 +1,6 @@
+# SE Linux policy database
+0       lelong  0xf97cff8c      SE Linux policy
+>16     lelong  x               v%d
+>20     lelong  1 		MLS
+>24     lelong  x       	%d symbols
+>28     lelong  x       	%d ocons
--- file-5.38/magic/Makefile.am~	2019-11-02 20:37:58.000000000 +0200
+++ file-5.38/magic/Makefile.am	2019-12-20 15:41:32.300192925 +0200
@@ -228,6 +228,7 @@
 $(MAGIC_FRAGMENT_DIR)/plan9 \
 $(MAGIC_FRAGMENT_DIR)/plus5 \
 $(MAGIC_FRAGMENT_DIR)/pmem \
+$(MAGIC_FRAGMENT_DIR)/policy \
 $(MAGIC_FRAGMENT_DIR)/polyml \
 $(MAGIC_FRAGMENT_DIR)/printer \
 $(MAGIC_FRAGMENT_DIR)/project \
