--- file-4.25/magic/Magdir/msdos.orig	2008-07-19 21:53:45.598043088 +0200
+++ file-4.25/magic/Magdir/msdos	2008-07-19 22:14:11.858041412 +0200
@@ -5,14 +5,13 @@
 
 # .BAT files (Daniel Quinlan, quinlan@yggdrasil.com)
 # updated by Joerg Jenderek at Oct 2008
-0	string	@			
->1	string/cW	\ echo\ off	DOS batch file text
+0	string/cW	@\ echo\ off	DOS batch file text
 !:mime	text/x-msdos-batch
->1	string/cW	echo\ off	DOS batch file text
+0	string/cW	@echo\ off	DOS batch file text
 !:mime	text/x-msdos-batch
->1	string/cW	rem\ 		DOS batch file text
+0	string/cW	@rem\ 		DOS batch file text
 !:mime	text/x-msdos-batch
->1	string/cW	set\ 		DOS batch file text
+0	string/cW	@set\ 		DOS batch file text
 !:mime	text/x-msdos-batch
 
 
@@ -329,7 +329,7 @@
 #0	byte		0xb8		COM executable
 0	uleshort&0x80ff	0x00b8		
 # modified by Joerg Jenderek
->1	lelong		!0x21cd4cff	COM executable for DOS
+#>1	lelong		!0x21cd4cff	COM executable for DOS
 # http://syslinux.zytor.com/comboot.php
 # (32-bit COMBOOT) programs *.C32 contain 32-bit code and run in flat-memory 32-bit protected mode
 # start with assembler instructions mov eax,21cd4cffh
@@ -334,27 +334,27 @@
 # GRR search is not working
 #2	search/28	\xcd\x21	COM executable for MS-DOS
 #WHICHFAT.cOM
-2	string	\xcd\x21		COM executable for DOS
+#2	string	\xcd\x21		COM executable for DOS
 #DELTREE.cOM DELTREE2.cOM
-4	string	\xcd\x21		COM executable for DOS
+#4	string	\xcd\x21		COM executable for DOS
 #IFMEMDSK.cOM ASSIGN.cOM COMP.cOM
-5	string	\xcd\x21		COM executable for DOS
+#5	string	\xcd\x21		COM executable for DOS
 #DELTMP.COm HASFAT32.cOM
-7	string	\xcd\x21		
->0	byte	!0xb8			COM executable for DOS
+#7	string	\xcd\x21		
+#>0	byte	!0xb8			COM executable for DOS
 #COMP.cOM MORE.COm
-10	string	\xcd\x21		
->5	string	!\xcd\x21		COM executable for DOS
+#10	string	\xcd\x21		
+#>5	string	!\xcd\x21		COM executable for DOS
 #comecho.com
-13	string	\xcd\x21		COM executable for DOS
+#13	string	\xcd\x21		COM executable for DOS
 #HELP.COm EDIT.coM
-18	string	\xcd\x21		COM executable for MS-DOS
+#18	string	\xcd\x21		COM executable for MS-DOS
 #NWRPLTRM.COm
-23	string	\xcd\x21		COM executable for MS-DOS
+#23	string	\xcd\x21		COM executable for MS-DOS
 #LOADFIX.cOm LOADFIX.cOm
-30	string	\xcd\x21		COM executable for MS-DOS
+#30	string	\xcd\x21		COM executable for MS-DOS
 #syslinux.com 3.11
-70	string	\xcd\x21		COM executable for DOS
+#70	string	\xcd\x21		COM executable for DOS
 # many compressed/converted COMs start with a copy loop instead of a jump
 0x6	search/0xa	\xfc\x57\xf3\xa5\xc3	COM executable for MS-DOS
 0x6	search/0xa	\xfc\x57\xf3\xa4\xc3	COM executable for DOS
