--- file-4.13.orig/doc/file.man
+++ file-4.13/doc/file.man
@@ -1,4 +1,4 @@
-.TH FILE __CSECTION__ "Copyright but distributable"
+.TH FILE __CSECTION__ "January 2004" "Copyrighted but distributable"
 .\" $Id$
 .SH NAME
 file
@@ -56,10 +56,9 @@
 meaning anything else (data is usually `binary' or non-printable).
 Exceptions are well-known file formats (core files, tar archives)
 that are known to contain binary data.
-When modifying the file
-.I __MAGIC__
-or the program itself, 
-.B "preserve these keywords" .
+When adding local definitions to
+.IR /etc/magic ,
+.BR "preserve these keywords" .
 People depend on knowing that all the readable files in a directory
 have the word ``text'' printed.
 Don't do as Berkeley did and change ``shell commands text''
@@ -98,7 +97,9 @@
 The concept of `magic number' has been applied by extension to data files.
 Any file with some invariant identifier at a small fixed
 offset into the file can usually be described in this way.
-The information identifying these files is read from the compiled
+The information identifying these files is read from
+.I /etc/magic
+and the compiled
 magic file
 .I __MAGIC__.mgc ,
 or 
@@ -339,12 +340,6 @@
 The order of entries in the magic file is significant.
 Depending on what system you are using, the order that
 they are put together may be incorrect.
-If your old
-.B file
-command uses a magic file,
-keep the old magic file around for comparison purposes
-(rename it to 
-.IR __MAGIC__.orig ).
 .SH EXAMPLES
 .nf
 $ file file.c file /dev/{wd0a,hda}
@@ -498,3 +493,7 @@
 .B ftp.astron.com
 in the directory
 .I /pub/file/file-X.YZ.tar.gz
+.PP
+This version contains some extensions from
+.B Debian
+(mainly new magic entries).
--- file-4.07.orig/src/Makefile.am
+++ file-4.07/src/Makefile.am
@@ -4,7 +4,7 @@
 
 bin_PROGRAMS = file
 
-AM_CPPFLAGS = -DMAGIC='"$(MAGIC)"'
+AM_CPPFLAGS = -DMAGIC='"/etc/magic:$(MAGIC)"'
 
 libmagic_la_SOURCES = magic.c apprentice.c softmagic.c ascmagic.c \
 	compress.c is_tar.c readelf.c print.c fsmagic.c \
--- file-4.07.orig/magic/Header
+++ file-4.07/magic/Header
@@ -1,5 +1,5 @@
-# Magic
 # Magic data for file(1) command.
-# Machine-generated from src/cmd/file/magdir/*; edit there only!
-# Format is described in magic(files), where:
-# files is 5 on V7 and BSD, 4 on SV, and ?? in the SVID.
+# Format is described in magic(5).
+# Don't edit this file, edit /etc/magic or send your suggested inclusions to
+# this file as a wishlist bug against file (using the reportbug utility).
+
--- file-4.11/magic/magic.mime.orig	2004-11-21 06:20:30.000000000 +0100
+++ file-4.11/magic/magic.mime	2004-11-22 23:30:02.976163088 +0100
@@ -190,7 +190,7 @@
 
 
 # MPEG Layer 3 sound files
-0       beshort		&0xffe0		audio/mpeg
+0       beshort&0xfffe  =0xfffa         audio/mpeg
 #MP3 with ID3 tag
 0	string		ID3		audio/mpeg
 # Ogg/Vorbis
@@ -289,10 +289,10 @@
 # because it tries to uncompress it to figure out what's inside.
 
 # standard unix compress
-0	string		\037\235	application/x-compress
+#0	string		\037\235	application/x-compress
 
 # gzip (GNU zip, not to be confused with [Info-ZIP/PKWARE] zip archiver)
-0       string          \037\213        application/x-gzip
+#0       string          \037\213        application/x-gzip
 
 0		string			PK\003\004		application/x-zip
 
@@ -393,19 +393,14 @@
 #
 # from Daniel Quinlan <quinlan@yggdrasil.com>
 #
-0	string		\<HEAD	text/html
-0	string		\<head	text/html
-0	string		\<TITLE	text/html
-0	string		\<title	text/html
-0       string          \<html	text/html
-0       string          \<HTML	text/html
+0	string/cB	\<!DOCTYPE\ html	text/html
+0	string/cb	\<head	text/html
+0	string/cb	\<title	text/html
+0       string/bc	\<html	text/html
 0	string		\<!--	text/html
-0	string		\<h1	text/html
-0	string		\<H1	text/html
-0	string		\<!doctype\ HTML	text/html
-0	string		\<!DOCTYPE\ HTML	text/html
-0	string		\<!doctype\ html	text/html
-0	string		\<!doctype\ HTML	text/html
+0	string/c	\<h1	text/html
+
+0	string		\<?xml			text/xml
 
 #------------------------------------------------------------------------------
 # images:  file(1) magic for image formats (see also "c-lang" for XPM bitmaps)
@@ -468,7 +463,7 @@
 0	beshort		0xffd8		image/jpeg
 
 # PC bitmaps (OS/2, Windoze BMP files)  (Greg Roelofs, newt@uchicago.edu)
-0	string		BM		image/bmp
+0	string		BM		image/x-ms-bmp
 #>14	byte		12		(OS/2 1.x format)
 #>14	byte		64		(OS/2 2.x format)
 #>14	byte		40		(Windows 3.x format)
@@ -778,6 +773,7 @@
 #
 0	string		FWS             
 >3	byte		x			application/x-shockwave-flash
+0	string		CWS			application/x-shockwave-flash
 
 # The following paramaters are created for Namazu.
 # <http://www.namazu.org/>
@@ -906,3 +902,7 @@
 # (made the mimetype up) 
 0	string	\0\0MMXPR3\0	application/x-quark-xpress-3
 
+# Gnumeric spreadsheet
+# This entry is only semi-helpful, as Gnumeric compresses its files, so
+# they will ordinarily reported as "compressed", but at least -z helps
+39      string          =<gmr:Workbook           application/x-gnumeric
--- file-4.07.orig/magic/magic.local
+++ file-4.07/magic/magic.local
@@ -0,0 +1,3 @@
+# Magic local data for file(1) command.
+# Insert here your local magic data. Format is described in magic(5).
+
