diff -ur file-5.19.orig/magic/Magdir/perl file-5.19/magic/Magdir/perl
--- file-5.19.orig/magic/Magdir/perl	2014-04-28 15:01:16.000000000 +0100
+++ file-5.19/magic/Magdir/perl	2014-10-07 06:10:40.780793722 +0100
@@ -18,7 +18,7 @@
 0	search/1	#!\ /usr/bin/env\ perl	Perl script text executable
 !:mime	text/x-perl
 0	search/1	#!
->0	regex	\^#!.*/bin/perl$	Perl script text executable
+>0	regex	\^#!.*/bin/perl([[:space:]].*)*$	Perl script text executable
 !:mime	text/x-perl
 
 # by Dmitry V. Levin and Alexey Tourbin
