diff -Nru file-4.02.bak/magic/Magdir/commands file-4.02/magic/Magdir/commands
--- file-4.02.bak/magic/Magdir/commands	Wed Apr  2 20:57:42 2003
+++ file-4.02/magic/Magdir/commands	Thu Apr  3 09:21:31 2003
@@ -13,8 +13,8 @@
 
 #
 # zsh/ash/ae/nawk/gawk magic from cameron@cs.unsw.oz.au (Cameron Simpson)
-0	string/b	#!\ /bin		Paul Falstad's zsh script text executable
-0	string/b	#!\ /usr/bin		Paul Falstad's zsh script text executable
+0	string/b	#!\ /bin/zsh		Paul Falstad's zsh script text executable
+0	string/b	#!\ /usr/bin/zsh	Paul Falstad's zsh script text executable
 0	string/b	#!\ /usr/local/bin/zsh	Paul Falstad's zsh script text executable
 0	string/b	#!\ /usr/local/bin/ash	Neil Brown's ash script text executable
 0	string/b	#!\ /usr/local/bin/ae	Neil Brown's ae script text executable
