--- file-5.26/src/file.h~	2016-03-31 20:51:12.000000000 +0300
+++ file-5.26/src/file.h	2016-04-17 12:37:55.562528440 +0300
@@ -421,7 +421,7 @@
 	uint16_t regex_max;
 	size_t bytes_max;		/* number of bytes to read from file */
 #define	FILE_INDIR_MAX			50
-#define	FILE_NAME_MAX			30
+#define	FILE_NAME_MAX			256
 #define	FILE_ELF_SHNUM_MAX		32768
 #define	FILE_ELF_PHNUM_MAX		2048
 #define	FILE_ELF_NOTES_MAX		256
